### PR TITLE
Reuse TextField Implementation

### PR DESCRIPTION
**Pull request by gtsiap**
Currently it's not easy to reuse `Components.TextInput`.

This PR introduces `FocusTextField` which contains a `UITextField` and an `AccessoryView`.
The idea is to have compassable views in order to create new components which will be needed.

These changes are source compatible for the consumers of `Components.TextInput`

### CODE REVIEW

**Comment by sergdort** [comment]
I it the same that @mluisbrown did [here](https://github.com/Babylonpartners/Bento/pull/115)?

Also could you please add screenshots

**Comment by sergdort** [comment]
@gtsiap is it a textfieid that highlights its borders when active?

**Comment by andersio** [comment]
Have you explored `UITextField.rightView`?

**Comment by mluisbrown** [comment]
> I it the same that @mluisbrown did [here](https://github.com/Babylonpartners/Bento/pull/115)?

@sergdort this is for TextField, my change it to MultilineTextInput.

**Comment by TheAdamBorek** [comment]
We were talking on #ios-oss slack channel of providing a way to compose multiple components by providing for "layout components" like
`StackViewComponent`(or `Row`/`Column` as Flutter use such naming - naming is not the case here).

Wouldn't that solve the problem you are working on?

**Comment by gtsiap** [comment]
> I it the same that @mluisbrown did [here](https://github.com/Babylonpartners/Bento/pull/115)?
> 
> Also could you please add screenshots

Its not tied to a specific style, that's why I haven't added any screenshots.
You can have a visual of it from here,
`Bento/Snapshots/ReferenceImages_64/BentoKitTests.TextInputSnapshotTests`

**Comment by gtsiap** [comment]
> @gtsiap is it a textfieid that highlights its borders when active?

Its design agnostic. Its not related to a specific stylesheet.

**Comment by gtsiap** [comment]
> Have you explored `UITextField.rightView`?

No, I didn't. I wanted to minimise the changes.

**Comment by gtsiap** [comment]
> We were talking on #ios-oss slack channel of providing a way to compose multiple components by providing for "layout components" like
> `StackViewComponent`(or `Row`/`Column` as Flutter uses such naming but naming is not the case here).
> 
> Wouldn't that solve the problem you are working on?

This discussion is related to this PR.

`StackViewComponent` or something equivalent wouldn't solve my issue because `Components.TextInput` has also a `contentView` which contains the textfield

File: BentoKit/BentoKit/Views/FocusTextField.swift [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                else { return "" }
+
+            return current.replacingCharacters(in: range, with: change)
+        }
+    }
+}
+
+extension FocusTextField {
+
+    public enum Accessory {
```
Comment (sergdort) [inline BentoKit/BentoKit/Views/FocusTextField.swift]: Why not use `AccessoryView.Accessory`?

**Comment by sergdort** [comment]
I think we should stop rushing and bounding unnecessary components and adding them into OSS library just because our design team requires it.
as @TheAdamBorek pointed out we can try to introduce `Row/Column` component (in BabylonUI) and see how it goes.

File: BentoKit/BentoKit/Views/FocusTextField.swift (hunk 1 above)
Comment (gtsiap) [reply BentoKit/BentoKit/Views/FocusTextField.swift]: The initial implementation was restraining  `AccessoryView.Accessory`.
I didn't want to change its behaviour, we can revise this decision.

**Comment by gtsiap** [comment]
I will close this PR for now until we figure out how we will handle compassable components.

**Pull request closed without merging**